src/viewers/VivViewer.js

```diff
@@ -107,14 +108,46 @@ export default class VivViewer extends PureComponent {
     return prevState;
   }
 
+  // eslint-disable-next-line consistent-return
+  onHover({ sourceLayer, coordinate, layer }) {
+    const { hoverHooks } = this.props;
+    const { handleValue } = hoverHooks;
+    if (handleValue) {
+      const { channelData, bounds } = sourceLayer.props;
+      const { zoom } = layer.context.viewport;
+      const { data, width } = channelData;
+      if (!coordinate) {
+        return null;
+      }
+      // Using floor means that as we zoom out, we are scaling by the zoom just passed, not the one coming.
+      const dataCoords = [
+        Math.floor(
+          (coordinate[0] - bounds[0]) / Math.max(1, 2 ** Math.floor(-zoom))
+        ),
+        Math.floor(
+          (coordinate[1] - bounds[3]) / Math.max(1, 2 ** Math.floor(-zoom))
+        )
+      ];
+      const hoverData = data.map(d => d[dataCoords[1] * width + dataCoords[0]]);
```

Comment:
Preference for doing math once, and named variables are probably more clear than tuples. Maybe something like:
```suggestion
      const intZoom = Math.max(1, 2 ** Math.floor(-zoom));
      const x = Math.floor((coordinate[0] - bounds[0]) / intZoom);
      const y = Math.floor((coordinate[1] - bounds[3]) / intZoom);
      const hoverData = data.map(d => d[y * width + x]);
```
... though I'm not sure `x` and `y` are right. A comment on the structure of the `bounds` array might also help.